devicetypes/heltun/heltun-hls01-switch.src/heltun-hls01-switch.groovy

```diff
@@ -0,0 +1,178 @@
+/**
+ *  Copyright 2021 Sarkis Kabrailian
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "HELTUN HLS01 Switch", namespace: "HELTUN", author: "Sarkis Kabrailian", cstHandler: true) {
+		capability "Energy Meter"
+		capability "Power Meter"
+		capability "Switch"
+		capability "Temperature Measurement"
+		capability "Voltage Measurement"
+		capability "Configuration"      
+		capability "Health Check" 
+		capability "Refresh"
+
+		fingerprint mfr: "0344", prod: "0004", inClusters:"0x5E,0x85,0x59,0x8E,0x55,0x86,0x72,0x5A,0x73,0x81,0x87,0x98,0x9F,0x6C,0x70,0x25,0x31,0x32,0x71,0x22,0x7A", deviceJoinName: "HELTUN Switch" //model: "000A"
+	}																		
+	preferences {
+		section { 
+			input (
+				type: "paragraph",
+				element: "paragraph",
+				title: "Parameter Configuration:",
+				description: "Set the Parameter ID and the corresponding Value to configure the parameters"
+			)
+
+            input (
+				name: "ParamId",
+				title: "Parameter ID:",
+				type: "number",
+				range: "0..65536",
+				required: false
+			)
+
+			input (
+				name: "ParamValue",
+				title: "Parameter Value:",
+				type: "number",
+				range: "0..2147483647",
+				required: false
+			)
```

Comment:
Please define each device setting explicitly.

Reply:
Could you please let me know what do you mean by "explicitly" ?

Reply:
Just make supported parameters visible (define them in the code, one by one) so users will not have to look for parameter IDs and parameter values in the manual.

Reply:
Ok, I am doing it.
I'm facing a problem, I would be grateful if you could help.
There are some parameters that are not supported in old firmware versions.
So I defined the firmware version for each parameter, the problem is that I cannot find a way to get the device firmware version and compare it with the defined version in the parameter.

Note that I'm able to get the firmware version of the device after it gets configured but that's too late.
Is there any way to do it ?

Reply:
I don't think so. This block is executed in the early stage, so there is no info about device model etc. (and it wouldn't be possible to get firmware version etc. ) 
I suggest to pick only common device settings that are crucial for this device.

Reply:
Done !